### PR TITLE
Use reusable GitHub Actions for testing

### PR DESCRIPTION
**Pull request by nofaralfasi**

### CODE REVIEW

File: hammer_cli_foreman_leapp.gemspec [hunk 1]
```diff
@@ -22,8 +22,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.test_files    = Dir['{test}/**/*']
 
-  spec.add_dependency 'hammer_cli_foreman', '>= 2.0.0', '< 4.0.0'
-  spec.add_development_dependency 'bundler'
-  spec.add_development_dependency 'rake'
-  spec.add_development_dependency 'rubocop', '~> 0.80'
+  spec.add_dependency 'hammer_cli_foreman', '~> 3.9'
+  spec.add_development_dependency 'bundler', '>= 2.4', '< 3'
```
Comment (nofaralfasi) [inline hammer_cli_foreman_leapp.gemspec]: Do we actually need it here?

Comment (stejskalleos) [reply hammer_cli_foreman_leapp.gemspec]: gandalf

Comment (stejskalleos) [reply hammer_cli_foreman_leapp.gemspec]: This was one of my first plugins, so all the changes made here were done without any actual knowledge.

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# Include: **/*.gemfile, **/Gemfile, **/gems.rb
+Bundler/OrderedGems:
+  Exclude:
+    - 'Gemfile'
+
+# Offense count: 2
+# This cop supports safe autocorrection (--autocorrect).
+# Configuration parameters: EnforcedStyle.
+# SupportedStyles: final_newline, final_blank_line
+Layout/TrailingEmptyLines:
```
Comment (ekohl) [inline .rubocop_todo.yml]: Please fix these, because this PR introduces them. Seeing how often they show up you may need to change a setting on your editor.

File: hammer_cli_foreman_leapp.gemspec [hunk 3]
```diff
@@ -22,8 +22,6 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
   spec.test_files    = Dir['{test}/**/*']
 
-  spec.add_dependency 'hammer_cli_foreman', '>= 2.0.0', '< 4.0.0'
-  spec.add_development_dependency 'bundler'
-  spec.add_development_dependency 'rake'
-  spec.add_development_dependency 'rubocop', '~> 0.80'
+  spec.add_dependency 'hammer_cli_foreman', '~> 3.9'
```
Comment (ekohl) [inline hammer_cli_foreman_leapp.gemspec]: This is pretty tight. It wouldn't work in nightly where it's now 3.11. Any particular reason to make it so tight?

**Review by ekohl — CHANGES_REQUESTED** [review]
Please check your editor for the line endings setting. Not sure which one you use, but in GitHub you can recognize the issue at the end of files by the circle with a dash in it.

File: .rubocop_todo.yml (hunk 2 above)
Comment (stejskalleos) [reply .rubocop_todo.yml]: I'm voting for fixing all the findings in this case.
It's not that big change in the end.

File: .rubocop.yml [hunk 4]
```diff
@@ -1,3 +1,11 @@
+inherit_from: .rubocop_todo.yml
+
+require:
+  - rubocop-performance
+
+AllCops:
+  TargetRubyVersion: 2.7
```
Comment (stejskalleos) [inline .rubocop.yml]: This should be 3.0, right?

File: .github/workflows/unit_tests.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Tests
+    uses: theforeman/actions/.github/workflows/test-gem.yml@v0
+    with:
+      command: bundle exec rake test
```
Comment (stejskalleos) [inline .github/workflows/unit_tests.yaml]: Trailing new line. As Ewoud said, check your editor to automatically add new line at the end

Comment (nofaralfasi) [reply .github/workflows/unit_tests.yaml]: I think we need to decide when we want to insert a new line and when it is unnecessary. See https://github.com/theforeman/hammer-cli-foreman-ansible/pull/27#discussion_r1496063705.

File: .rubocop.yml (hunk 4 above)
Comment (ekohl) [reply .rubocop.yml]: No, we haven't dropped Ruby 2.7 yet.

File: .github/workflows/unit_tests.yaml (hunk 5 above)
Comment (ekohl) [reply .github/workflows/unit_tests.yaml]: > I think we need to decide when we want to insert a new line and when it is unnecessary.

In UNIX files every line has a line ending, but in that case you had an additional empty line. Most editors have a setting to enforce this and many do that by default.

File: hammer_cli_foreman_leapp.gemspec (hunk 3 above)
Comment (ekohl) [reply hammer_cli_foreman_leapp.gemspec]: Oh, I just realized that this will work. Disregard my comment.

File: .rubocop.yml (hunk 4 above)
Comment (nofaralfasi) [reply .rubocop.yml]: We can use:
```
AllCops:
  TargetRubyVersion: 2.7..3.0
```

Comment (stejskalleos) [reply .rubocop.yml]: This is the way IMHO

Comment (ekohl) [reply .rubocop.yml]: No, you set a minimum version. https://docs.rubocop.org/rubocop/configuration.html#setting-the-target-ruby-version states:
> Users may let RuboCop know the oldest version of Ruby which your project supports with:

Though there's also a line that says it will use the gemspec if it has `required_ruby_version` so perhaps it's redundant and we can drop this line altogether. If that works, I'd prefer that because I dislike duplicating information.

File: Rakefile [hunk 6]
```diff
@@ -5,7 +5,24 @@ require 'bundler/gem_tasks'
 
 Bundler::GemHelper.install_tasks
 
-require "hammer_cli_foreman_leapp/version"
-require "hammer_cli_foreman_leapp/i18n"
-require "hammer_cli/i18n/find_task"
+Rake::TestTask.new do |t|
+  t.libs.push 'lib'
+  t.verbose = true
+end
```
Comment (ekohl) [inline Rakefile]: Why was this needed?

**Review by ekohl — COMMENTED** [review]
Looks good, one last question.

File: Rakefile (hunk 6 above)
Comment (nofaralfasi) [reply Rakefile]: How else would you define a Rake task for running the tests?

**Comment by nofaralfasi** [comment]
@ShimShtein can you please review?

**Comment by stejskalleos** [comment]
LGTM, @ekohl can we merge?

File: Rakefile (hunk 6 above)
Comment (ekohl) [reply Rakefile]: There aren't any tests so the task is not needed. I think this repository should stick with just using Rubocop for now, until tests are written.

File: .github/workflows/unit_tests.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
+  test:
+    name: Tests
+    uses: theforeman/actions/.github/workflows/test-gem.yml@v0
```
Comment (ekohl) [inline .github/workflows/unit_tests.yaml]: There are no tests, so this can be dropped
```suggestion
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — APPROVED** [review]
